doc/specs/#16599 - Quick Fix.md

```diff
@@ -0,0 +1,56 @@
+---
+author: Carlos Zamora @carlos-zamora
+created on: 2024-04-01
+last updated: 2024-04-05
+issue id: #16599
+---
+# Quick Fix
+## Solution Design
+Quick Fix will be a new UI surface that allows the user to interact with the terminal and leverage the context of a specific command being executed. The UI is discussed in the [UI/UX Design section](#ui/ux-design).
+
+The UI itself will live in the TerminalControl layer, but it can be populated by both the TerminalControl and the TerminalApp layer (similar to the context menu). The suggestions will be appended as top-level items, but a nested design can be revisited as more sources are added to populate the Quick Fix menu. Only one Quick Fix menu will be available at a time as it only appears on the latest prompt if an error occurred; this is in alignment with VS Code's design.
+
+The actual suggestions will be populated from various sources such as the following:
+- WinGet Command Not Found OSC Sequence:
+	- `OSC 9001; CmdNotFound; <missingCmd>` will be used by the attached CLI app to notify the terminal that a command `missingCmd` wasn't found. The TerminalControl layer will notify the TerminalApp layer that this VT sequence as received and the TerminalApp layer will use WinGet's API to search for any packages that would remediate this error and construct a list of suggested packages to be installed.
+`OSC 9001` can be expanded on to add more suggestions via VT sequences from the underlying CLI app; for more on this, read the [Future considerations section](#future-considerations).
+
+TerminalApp will be notified upon receiving an `OSC 9001; CmdNotFound`. At this point, we'll know that a Quick Fix menu must be shown at the next prompt, and we can use the WinGet API to convert the given `missingCmd` into a list of suggestions (in this case, of the form `winget install...`). These suggestions will be handed back down to the TerminalControl layer to be added to the Quick Fix menu as entries. Since we're in the TerminalControl layer, the suggestions may not need to be converted into SendInput actions, as we already have access to the `RawWriteString()` api. The menu will simply use `RawWriteString()` to inject the string into the prompt.
```

Comment:
I'm a little curious what the shape of this API actually looks like? Is it like, 

```c#
runtimeclass TermControl {
    // ...
    event Windows.Foundation.TypedEventHandler<TermControl, CommandNotFoundRequestedArgs> CommandNotFound;
    Microsoft.UI.Xaml.Controls.CommandBarFlyout QuickFixMenu { get; };
}
```

(maybe not a CommandBarFlyout - something that's appropriate. Flyouts are hard)

...

```c++
// TerminalPage.cpp
term.CommandNotFound({ this, &TerminalPage::DoAWingetLookupForCommand });
term.QuickFixMenu().Opening([this](auto&& sender, auto&& /*args*/) {
     this->AddQuickFixesToMenu(term.QuickFixMenu());
});
```